Source/WebCore/page/Performance.cpp

```diff
@@ -375,13 +376,19 @@ void Performance::setResourceTimingBufferSize(unsigned size)
 
 void Performance::reportFirstContentfulPaint(DOMHighResTimeStamp timestamp)
 {
+    if (RefPtr context = scriptExecutionContext())
+        InspectorInstrumentation::didEnqueuePerformanceEntry(*context, "first-contentful-paint"_s);
+
     ASSERT(!m_firstContentfulPaint);
     m_firstContentfulPaint = PerformancePaintTiming::createFirstContentfulPaint(timestamp);
     queueEntry(*m_firstContentfulPaint);
 }
 
 void Performance::enqueueLargestContentfulPaint(Ref<LargestContentfulPaint>&& paintEntry)
 {
+    if (RefPtr context = scriptExecutionContext())
+        InspectorInstrumentation::didEnqueuePerformanceEntry(*context, "largest-contentful-paint"_s, paintEntry->element(), paintEntry->size());
```

Comment:
NIT: in order to avoid doing extra work when Web Inspector is not open, id suggest just passing either the entire `Performance` object or the `ScriptExecutionContext*`
```cpp
InspectorInstrumentation::enqueueLargestContentfulPaint(*this, paintEntry);
```
this way, we can delay the work of unpacking anything until we actually get to the `InspectorTimelineAgent`, which will only happen if Web Inspector is open (see `FAST_RETURN_IF_NO_FRONTENDS`)